homeassistant/components/plex/server.py

```diff
@@ -0,0 +1,59 @@
+"""Shared class to maintain Plex server instances."""
+import logging
+import plexapi.server
+from requests import Session
+
+from homeassistant.const import CONF_TOKEN, CONF_URL, CONF_VERIFY_SSL
+
+from .const import DEFAULT_VERIFY_SSL
+
+_LOGGER = logging.getLogger(__package__)
+
+
+class PlexServer:
+    """Manages a single Plex server connection."""
+
+    def __init__(self, server_config):
+        """Initialize a Plex server instance."""
+        self._plex_server = None
+        self._url = server_config.get(CONF_URL)
+        self._token = server_config.get(CONF_TOKEN)
+        self._verify_ssl = server_config.get(CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL)
+
+    def connect(self):
+        """Connect to a Plex server directly, obtaining direct URL if necessary."""
+
+        def _connect_with_url():
```

Comment:
Should we just inline this function into `connect`?

Reply:
Future PRs will add another function here.